hash_practice/exercises.py

```diff
@@ -2,18 +2,45 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)
     """
-    pass
+    # start with empty dictionary

```

Comment:
I won't put out every one, but as a style thing, you generally want to get rid of "what" comments like these just out of style concerns. They are fine for when you are scaffolding out your code, but they tend to have the problem of falling out of sync with the actual implementation, which is usually good enough to tell a reader what the code does.

Much more useful are "why" comments that tell the reader why a code does something a certain way (likely because you had to do it some weird messy way).

Reply:
There's other places that you have such comments, but I won't point out every one.